internal/services/aadgraph/application_app_role_resource.go

```diff
@@ -0,0 +1,259 @@
+package aadgraph
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/terraform-providers/terraform-provider-azuread/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azuread/internal/services/aadgraph/graph"
+	"github.com/terraform-providers/terraform-provider-azuread/internal/tf"
+	"github.com/terraform-providers/terraform-provider-azuread/internal/utils"
+	"github.com/terraform-providers/terraform-provider-azuread/internal/validate"
+)
+
+func ApplicationAppRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: applicationAppRoleResourceCreate,
+		Update: applicationAppRoleResourceUpdate,
+		Read:   applicationAppRoleResourceRead,
+		Delete: applicationAppRoleResourceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"application_object_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.UUID,
+			},
+
+			"allowed_member_types": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice(
+						[]string{"User", "Application"},
+						false,
+					),
+				},
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"display_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"is_enabled": {
```

Comment:
Do you think we should do this for the upcoming msgraph resources? If we change this now, it would also make sense to change the corresponding block attribute in the application resource - which we could do but that would break existing configs. Thoughts?

Reply:
good point - yeah let's fix this going forwards in the new resources, more a heads up for now 👍